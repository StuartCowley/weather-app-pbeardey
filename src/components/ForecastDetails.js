// src/componants/ForecastDetails.js

import React from "react";
import PropTypes from "prop-types";

function ForecastDetails({forecast}) {
//   const { forecast } = props;
  const { date, humidity, temperature, wind } = forecast;
  const formattedDate = new Date(date).toDateString();

  return (
    <div className="forecast-datails" data-testid="forecast-details">
      <div className="forecast-details__date">{formattedDate}</div>
      <div
        className="forecast-details__temperature"
        data-testid="forecast-details__temp-max"
      >
        Max Temperature: {temperature.max}&deg;C
      </div>
      <div
        className="forecast-details__temperature"
        data-testid="forecast-details__temp-min"
      >
        Min Temperature: {temperature.min}&deg;C
      </div>
      <div
        className="forecast-details__humidity"
        data-testid="forecast-details__humidity"
      >
        Humidity: {humidity}%
      </div>
      <div className="forecast-wind" data-testid="forecast-details__wind">
        Wind: {wind.speed}mph {wind.direction}
      </div>
    </div>
  );
}

ForecastDetails.propTypes = {
  forecast: PropTypes.shape({
    date: PropTypes.number,
    description: PropTypes.string,
    humidity: PropTypes.number,
    icon: PropTypes.string,
    temperature: PropTypes.shape({
      max: PropTypes.number,
      min: PropTypes.number,
    }),
    wind: PropTypes.shape({
      direction: PropTypes.string,
      speed: PropTypes.number,
    }),
  }).isRequired,
};

export default ForecastDetails;
