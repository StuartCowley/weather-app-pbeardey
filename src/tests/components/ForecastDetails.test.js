// src/test/components/ForecastDetails.test.js

import React from 'react';
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("ForecastDetails", () => {
  const validProps = {
    date: 1111111,
    temperature: {
      min: 12,
      max: 22,
    },
    humidity: 80,
    wind: {
      direction: "nnw",
      speed: 12,
    },
    onForecastSelect() {
      return 1;
    },
  };

  describe("ForecastDetails-snapshot", () => {
    it("renders the 4 propss correctly", () => {
      const { asFragment } = render(<ForecastDetails forecast={validProps} />);
      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe("ForecastDetails-values", () => {
    const { getByText } = render(<ForecastDetails forecast={validProps} />);
    expect(getByText("Thu Jan 01 1970")).toHaveAttribute(
      "class",
      "forecast-details__date"
    );
    expect(getByText("Max Temperature: 22°C")).toHaveAttribute(
      "Class",
      "forecast-details__temperature"
    );
    expect(getByText("Min Temperature: 12°C")).toHaveAttribute(
      "Class",
      "forecast-details__temperature"
    );
    expect(getByText("Humidity: 80%")).toHaveAttribute(
      "Class",
      "forecast-details__humidity"
    );
  });
});