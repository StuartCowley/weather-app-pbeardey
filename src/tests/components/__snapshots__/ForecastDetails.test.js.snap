// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails ForecastDetails-snapshot renders the 4 propss correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-datails"
    data-testid="forecast-details"
  >
    <div
      class="forecast-details__date"
    >
      Thu Jan 01 1970
    </div>
    <div
      class="forecast-details__temperature"
      data-testid="forecast-details__temp-max"
    >
      Max Temperature: 22°C
    </div>
    <div
      class="forecast-details__temperature"
      data-testid="forecast-details__temp-min"
    >
      Min Temperature: 12°C
    </div>
    <div
      class="forecast-details__humidity"
      data-testid="forecast-details__humidity"
    >
      Humidity: 80%
    </div>
    <div
      class="forecast-wind"
      data-testid="forecast-details__wind"
    >
      Wind: 12mph nnw
    </div>
  </div>
</DocumentFragment>
`;
